File: cpp/leetcode/225.cpp
class Stack {
public:
	queue<int> q;
	void push(int x) {
		q.push(x);
		for(int i = 0; i < q.size()-1; ++i){
			q.push(q.front());
			q.pop();
		}
	}

	void pop() {
		q.pop();
	}

	int top() {
		return q.front();
	}

	bool empty() {
		return q.empty();
	}
};

File: python/leetcode/225.py
class Stack:
    def __init__(self):
        self.stack = collections.deque([])

    def push(self, x):
        self.stack.append(x)

    def pop(self):
        for i in range(len(self.stack) - 1):
            self.stack.append(self.stack.popleft())
        self.stack.popleft()

    def top(self):
        return self.stack[-1]

    def empty(self):
        return len(self.stack) == 0